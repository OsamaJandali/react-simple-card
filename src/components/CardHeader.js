// @flow

import React from "react";

type CardHeader = {
	children: any
};

export default ({ children }: CardHeader) => (
	<div className="card__header">{children}</div>
);
