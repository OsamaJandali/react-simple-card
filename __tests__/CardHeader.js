// @flow

import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardHeader from "../src/components/CardHeader";

it("renders CardHeader correctly", () => {
	const wrapper = shallow(
		<CardHeader>
			Quisque eget maximus tortor. Donec tortor est, placerat quis.
		</CardHeader>
	);
	expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders CardHeader correctly with extra styles", () => {
	const wrapper = shallow(
		<CardHeader style={{ backgroundColor: "green" }}>
			Quisque eget maximus tortor. Donec tortor est, placerat quis.
		</CardHeader>
	);
	expect(toJson(wrapper)).toMatchSnapshot();
});
