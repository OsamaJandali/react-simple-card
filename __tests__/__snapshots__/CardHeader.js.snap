// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardFooter correctly 1`] = `
<div
  className="card__footer"
  style={
    Object {
      "alignItems": "center",
      "borderTop": "1px solid #ccc",
      "display": "flex",
      "marginTop": "0",
      "padding": "10px 18px",
    }
  }
>
  Quisque eget maximus tortor. Donec tortor est, placerat quis.
</div>
`;
